src/stream/src/common/builder.rs

```diff
@@ -115,10 +138,14 @@ impl StreamChunkBuilder {
     ) -> ArrayResult<Option<StreamChunk>> {
         self.ops.push(op);
         for (i, d) in row_update.values().enumerate() {
-            self.column_builders[i + self.update_start_pos].append_datum_ref(d)?;
+            if let Some(idx) = self.index_mapping[i + self.update_start_pos] {
+                self.column_builders[idx].append_datum_ref(d)?;
+            }
         }
         for (i, d) in row_matched.values().enumerate() {
-            self.column_builders[i + self.matched_start_pos].append_datum(d)?;
+            if let Some(idx) = self.index_mapping[i + self.matched_start_pos] {
+                self.column_builders[idx].append_datum(d)?;
+            }
         }
```

Comment:
Can we do something like
```rust
for i, idx in self.output_indices_update {
    self.column_builders[i].append_datum_ref(row_update.value_at(idx)?
}

// same for the matched side
```
Essentially we may want `output_indices_update` to replace `update_start_pos`.
This may increase the performance if there are many columns.

Reply:
I did something similar to this in recent commits by only looping through `output_indices`,  though it is slightly different from your suggestion. Can you take a look at this?